src/healthkit.cs

```diff
@@ -2442,4 +2557,220 @@ interface HKWorkoutRouteQuery {
 		IntPtr Constructor (HKWorkoutRoute workoutRoute, HKWorkoutRouteBuilderDataHandler dataHandler);
 	}
 
+	[Watch (5,0), iOS (12,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface HKWorkoutBuilder
+	{
+		[NullAllowed, Export ("device", ArgumentSemantic.Copy)]
+		HKDevice Device { get; }
+
+		[NullAllowed, Export ("startDate", ArgumentSemantic.Copy)]
+		NSDate StartDate { get; }
+
+		[NullAllowed, Export ("endDate", ArgumentSemantic.Copy)]
+		NSDate EndDate { get; }
+
+		[Export ("workoutConfiguration", ArgumentSemantic.Copy)]
+		HKWorkoutConfiguration WorkoutConfiguration { get; }
+
+		[Internal]
+		[Export ("metadata", ArgumentSemantic.Copy)]
+		NSDictionary _Metadata { get; }
+
+		[Wrap ("_Metadata")]
+		HKMetadata Metadata { get; }
+
+		[Export ("workoutEvents", ArgumentSemantic.Copy)]
+		HKWorkoutEvent[] WorkoutEvents { get; }
+
+		[Export ("initWithHealthStore:configuration:device:")]
+		IntPtr Constructor (HKHealthStore healthStore, HKWorkoutConfiguration configuration, [NullAllowed] HKDevice device);
+
+		[Async]
+		[Export ("beginCollectionWithStartDate:completion:")]
+		void BeginCollection (NSDate startDate, Action<bool, NSError> completionHandler);
```

Comment:
There are a bunch of `Action<bool, NSError>` what does the `bool` means? Can't we have a named delegate here instead?

Reply:
@dalexsoto all those, are following the pattern already  found in the file, I personally would follow the same pattern in all possible cases. @spouliot @VincentDondain do you have any comments since you work on this before?